day_two/src/main.rs

```diff
@@ -0,0 +1,79 @@
+use std::fs;
+
+fn main() {
+    let raw_input = fs::read_to_string("/Users/phil/Developer/aoc2024/day_two/src/input.txt")
+        .expect("Should have been able to read the file");
+
+    let lines = raw_input.lines();
+    let mut line_vectors: Vec<Vec<i32>> = Vec::new();
+    for line in lines {
+        let split_line: Vec<i32> = line
+            .split_whitespace()
+            .map(|val| val.parse().expect("Failed to get number"))
+            .collect();
+
+        line_vectors.push(split_line);
+    }
+
+    println!(
+        "{}",
+        line_vectors
+            .iter()
+            .filter(|line| is_damp_safe(&line))
```

Comment:
Slightly clearner to use `into_iter`: https://stackoverflow.com/questions/34733811/what-is-the-difference-between-iter-and-into-iter

```rust
line_vectors.into_iter().filter(is_damp_safe).count()
```

You don't need to borrow `&line` at all in the filter in the way you did it, but in a case like this `into_iter` just moves the items into the new scope